editor/src/clj/editor/cljfx_form_view.clj

```diff
@@ -93,32 +94,57 @@
 (defmethod handle-event :set [{:keys [path fx/event]}]
   {:set [path event]})
 
-(def uri-string-converter
+(def ^:private uri-string-converter
   (proxy [StringConverter] []
     (toString
       ([] "uri-string-converter")
       ([v] (str v)))
     (fromString [v]
       (url/try-parse v))))
 
-(def number-converter
+(def ^:private float-converter
+  (proxy [StringConverter] []
+    (toString
+      ([] "float-string-converter")
+      ([v] (field-expression/format-real v)))
+    (fromString [v]
+      (or (field-expression/to-float v)
+          (throw (RuntimeException.))))))
+
+(def ^:private double-converter
   (proxy [StringConverter] []
     (toString
-      ([] "number-string-converter")
-      ([v] (field-expression/format-number v)))
+      ([] "double-string-converter")
+      ([v] (field-expression/format-real v)))
     (fromString [v]
       (or (field-expression/to-double v)
           (throw (RuntimeException.))))))
 
-(def int-converter
+(def ^:private int-converter
   (proxy [StringConverter] []
     (toString
       ([] "int-string-converter")
       ([v] (field-expression/format-int v)))
     (fromString [v]
-      (or (int (field-expression/to-double v))
+      (or (field-expression/to-int v)
+          (throw (RuntimeException.))))))
+
+(def ^:private long-converter
+  (proxy [StringConverter] []
+    (toString
+      ([] "long-string-converter")
+      ([v] (field-expression/format-int v)))
+    (fromString [v]
+      (or (field-expression/to-long v)
           (throw (RuntimeException.))))))
 
+(defn- number-string-converter [value]
+  (condp instance? value
+    Float float-converter
+    Double double-converter
+    Integer int-converter
+    Long long-converter))
+
```

Comment:
As you can see, the changes to this file are mainly related to preserving data types when editing.